### PR TITLE
Job names should include the image type

### PR DESCRIPTION
**Pull request by prashantchoudhary**
https://github.com/YannickRe/azuredevops-buildagents/issues/81

### CODE REVIEW

File: buildagent-generation-update-vmss.yml [hunk 1]
```diff
@@ -25,7 +29,7 @@ jobs:
 
   variables:
   - name: ManagedImageName
-    value: $[stageDependencies.buildagent_template_vm.generate_image.outputs['setVars.ManagedImageNameOutput'] ]
+    value: $[stageDependencies.${{ parameters.build_stage_job_name }}.outputs['setVars.ManagedImageNameOutput'] ]
```
Comment (YannickRe) [inline buildagent-generation-update-vmss.yml:32]: Can we add the `.generate_image` back to this line? It's a bit cleaner so we aren't passing that from multiple places.

File: buildagent-generation-update-vmss.yml [hunk 2]
```diff
@@ -14,6 +14,10 @@ parameters:
 - name: vmss_names
   displayName: VmssNames
   type: string
+- name: build_stage_job_name
+  displayName: Name of the stage and job that builds the image
+  type: string
+  default: buildagent_template_vm.generate_image
```
Comment (YannickRe) [inline buildagent-generation-update-vmss.yml:20]: Can we remove `.generate_image` here?

File: buildagent-generation-template.yml [hunk 3]
```diff
@@ -279,4 +280,6 @@ stages:
           variable_group_name: ${{ parameters.variable_group_name }}
           agent_pool: ${{ parameters.agent_pool }}
           repository_base_path: ${{ parameters.repository_base_path }}
-          vmss_names: $(VmssNames)
\ No newline at end of file
+          vmss_names: $(VmssNames)
+          build_stage_job_name: buildagent_template_vm_${{ parameters.image_type }}.generate_image
```
Comment (YannickRe) [inline buildagent-generation-template.yml:284]: Can we remove `.generate_image` here?

File: buildagent-generation-template.yml [hunk 4]
```diff
@@ -243,22 +243,23 @@ stages:
...
     dependsOn:
-      - buildagent_template_vm
+      - buildagent_template_vm_${{ parameters.image_type }}
     jobs:
       - template: buildagent-generation-galleryvm.yml
         parameters:
           variable_group_name: ${{ parameters.variable_group_name }}
           agent_pool: ${{ parameters.agent_pool }}
           repository_base_path: ${{ parameters.repository_base_path }}
+          build_stage_job_name: buildagent_template_vm_${{ parameters.image_type }}.generate_image
```
Comment (YannickRe) [inline buildagent-generation-template.yml:257]: Can we remove `.generate_image` here?

File: buildagent-generation-galleryvm.yml [hunk 5]
```diff
@@ -11,20 +11,24 @@ parameters:
   displayName: Scripts Path
   type: string
   default: .
+- name: build_stage_job_name
+  displayName: Name of the stage and job that builds the image
+  type: string
+  default: buildagent_template_vm.generate_image
```
Comment (YannickRe) [inline buildagent-generation-galleryvm.yml:17]: Can we remove `.generate_image` here?

File: buildagent-generation-galleryvm.yml [hunk 6]
```diff
@@ -11,20 +11,24 @@ parameters:
...
   timeoutInMinutes: 600
   cancelTimeoutInMinutes: 90
-  pool: 
+  pool:
     name: ${{ parameters.agent_pool }}
     
   variables:
   - name: GalleryVmImageDefinition
-    value: $[stageDependencies.buildagent_template_vm.generate_image.outputs['setVars.GalleryVmImageDefinition'] ]
+    value: $[stageDependencies.${{parameters.build_stage_job_name}}.outputs['setVars.GalleryVmImageDefinition'] ]
```
Comment (YannickRe) [inline buildagent-generation-galleryvm.yml:29]: Can we add the `.generate_image` back to this line? It's a bit cleaner so we aren't passing that from multiple places.

File: buildagent-generation-galleryvm.yml [hunk 7]
```diff
@@ -11,20 +11,24 @@ parameters:
...
   - name: ManagedImageName
-    value: $[stageDependencies.buildagent_template_vm.generate_image.outputs['setVars.ManagedImageNameOutput'] ]
+    value: $[stageDependencies.${{parameters.build_stage_job_name}}.outputs['setVars.ManagedImageNameOutput'] ]
```
Comment (YannickRe) [inline buildagent-generation-galleryvm.yml:31]: Can we add the `.generate_image` back to this line? It's a bit cleaner so we aren't passing that from multiple places.

**Review by YannickRe — CHANGES_REQUESTED** [review]
I have added comments, could you check them? Thanks for the work you did!

**Review by YannickRe — APPROVED** [review]

**Comment by YannickRe** [comment]
I'm sorry, this took me a while as I had to make the requested changes myself.
Finally got around to it, and now I could merge.

**Comment by prashantchoudhary** [comment]
Thanks a lot. I got busy with something else too and it just slipped my mind.

**Pull request merged**